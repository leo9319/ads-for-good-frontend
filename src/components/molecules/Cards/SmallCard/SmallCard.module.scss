@import '../../../../styles/colours';

.container {
  width: 100%;
  height: 100%;
  padding: var(--spacing-5xl);
  background-color: $color-neutral-100;
  box-shadow: 0px 1px 3px 0px $color-box-shadow;

  --card-border-width: 0;
  --card-border-radius: var(--spacing-xl);
  --card-background-color: none;

  &.size-2 {
    padding: var(--spacing-2xl);
    --card-border-radius: var(--spacing-xl);
  }

  &.size-3 {
    padding: var(--spacing-md);
    --card-border-radius: var(--spacing-md);
  }

  &.inverse {
    background-color: $color-white;
  }
}
