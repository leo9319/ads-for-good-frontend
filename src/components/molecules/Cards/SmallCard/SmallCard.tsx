import React, { ReactElement, ReactNode } from 'react';
import Card from '@radix-styles/atoms/Card';
import { classNames } from '@utils/common/classNames';
import styles from './SmallCard.module.scss';

/**
 * @interface
 * The Small Card props type of {@link SmallCard}
 */
export interface SmallCardProps {
  /**
   * Children as jsx, react component will show inside the modal
   * @property
   */
  children: ReactNode;
  /**
   * Determines the size of the Small Card
   * @defaultValue `2`
   * @property
   */
  size?: '1' | '2' | '3';
  /**
   * Controls the visual style of the Small Card
   * @defaultValue `primary`
   * @property
   */
  mode?: 'primary' | 'inverse';
  /**
   * Additional class names for customizing styles
   * @property
   */
  containerClass?: string;
}

/**
 * The Small Card is a structured component that organizes related content
 * and supporting customizable sizes and visual variants as well.
 *
 * @group Molecules
 * @category Component
 *
 * @param {SmallCardProps} props - The Small Card props
 * @returns {ReactElement} - Small Card Component
 *
 */
export const SmallCard = ({
  children,
  containerClass,
  size = '2',
  mode = 'primary',
}: SmallCardProps): ReactElement => {
  return (
    <Card
      className={classNames(
        styles.container,
        styles[`size-${size}`],
        styles[mode],
        containerClass
      )}
    >
      {children}
    </Card>
  );
};

export default SmallCard;
