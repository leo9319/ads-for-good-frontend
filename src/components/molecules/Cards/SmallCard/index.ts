export type { SmallCardProps } from './SmallCard';
export { default } from './SmallCard';
export { SmallCard } from './SmallCard';
