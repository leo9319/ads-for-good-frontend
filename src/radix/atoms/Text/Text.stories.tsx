import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import { Text, TextProps } from './Text';
import { Theme } from '@radix-ui/themes';

const meta: Meta<typeof Text> = {
  title: 'Library/Atoms/Text',
  component: Text,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  decorators: [
    Story => (
      <Theme>
        <Story />
      </Theme>
    ),
  ],
  argTypes: {
    children: {
      control: 'text',
    },
    size: {
      description: 'Font size',
      control: {
        type: 'select',
      },
      options: [
        '1',
        '2',
        '3',
        '4',
        '5',
        '6',
        '7',
        '8',
        '9',
        '10',
        '11',
        '12',
        '13',
      ],
    },
    weight: {
      control: {
        type: 'select',
      },
      options: ['regular', 'semibold', 'bold'],
    },
    className: {
      control: 'text',
    },
  },
  args: {
    children: 'Sample Text',
    size: '3',
    weight: 'regular',
    as: 'span',
  },
};

export default meta;

type Story = StoryObj<TextProps>;

export const Default: Story = {
  args: {
    children: 'Default Text',
  },
};

export const LargeBoldText: Story = {
  args: {
    children: 'Large Bold Text',
    size: '7',
    weight: 'bold',
  },
};

export const TextWithHTML: Story = {
  args: {
    children:
      '<div><p>Hello, world!</p><span>React is great!</span><ul><li>List 1</li><li>List 2</li><li>List 3</li><li>List 4 <a href="#">some link1</a></li></ul></div>',
    size: '2',
    weight: 'regular',
  },
};
