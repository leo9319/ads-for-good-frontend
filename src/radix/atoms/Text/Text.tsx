import { TextSize, TextWeight } from '@internal/types/common/style';
import { Text as RadixText, TextProps as Props } from '@radix-ui/themes';
import { getFontStyles } from '@utils/elements/font';
import React, { Fragment, ReactElement } from 'react';
import { useContent } from '@utils/hooks';
import { Config } from '@utils/common/htmlToRadix';
import { hasTags } from '@utils/common/text';

export type TextProps = Omit<Props, 'size' | 'weight'> & {
  /**
   * The size of the text
   * @property
   */
  size?: TextSize | null;
  /**
   * The weight of the text
   * @property
   */
  weight?: TextWeight | null;
  /**
   * config for the text
   * @property
   */
  config?: TextProps;
};

/**
 * Text component is to apply the text style to the children element.
 *
 * @group Atoms
 * @category Radix Component
 *
 * @param {TextProps} props - The Text props
 * @returns {ReactElement} - Text Component
 */
export const Text = ({
  children,
  size = '3',
  weight = 'regular',
  className,
  config,
  ...props
}: TextProps): ReactElement => {
  const classes = getFontStyles(props, size, weight, className);
  const newProps = { className: classes, ...(props as Props) };

  const defaultTextConfig = {
    size,
    weight,
  };

  const textConfig: Config = {
    textProps: {
      ...(config ?? defaultTextConfig),
    },
  };

  const hasStringAsChildren = typeof children === 'string';
  const { id, newChildren } = useContent(
    hasStringAsChildren ? children : null,
    textConfig
  );

  if (hasStringAsChildren) {
    return (
      <RadixText
        id={id}
        {...newProps}
        asChild={props.asChild || hasTags(children)}
      >
        <Fragment>{newChildren}</Fragment>
      </RadixText>
    );
  }
  return <RadixText {...newProps}>{children}</RadixText>;
};

export default Text;
