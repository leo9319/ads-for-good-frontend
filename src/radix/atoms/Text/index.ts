export { Text } from './Text';
export { default } from './Text';
export type { TextProps } from './Text';
