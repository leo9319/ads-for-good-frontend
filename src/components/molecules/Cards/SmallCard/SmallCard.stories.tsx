import React from 'react';
import SmallCard from './SmallCard';
import type { Meta, StoryObj } from '@storybook/react';
import LoginForm from '@components/molecules/LoginForm';
import Theme from '@radix-styles/atoms/Theme';

// Common Children for Small Card wrapper stories
const children = (
  <LoginForm
    emailPlaceholder="Email address"
    onSubmit={() => {}}
    passwordPlaceholder="Password"
    text="Login"
  />
);

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: 'Library/Molecules/Cards/SmallCard',
  component: SmallCard,
  decorators: [
    Story => (
      <Theme>
        <Story />
      </Theme>
    ),
  ],
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
  argTypes: {
    children: {
      control: { type: 'object' },
    },
    size: { control: 'select', options: ['1', '2', '3'] },
    mode: { control: 'select', options: ['primary', 'inverse'] },
    containerClass: { control: 'text' },
  },
  // more on args: https://storybook.js.org/docs/writing-stories/args#args-object
  args: {
    size: '2',
    children: children,
  },
} satisfies Meta<typeof SmallCard>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args

export const Primary: Story = {
  args: {
    mode: 'primary',
  },
};

export const Inverse: Story = {
  args: {
    mode: 'inverse',
  },
};
