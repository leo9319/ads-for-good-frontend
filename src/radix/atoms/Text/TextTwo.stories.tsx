import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import { Text, TextProps } from './Text';
import { Theme } from '@radix-ui/themes';

const meta: Meta<typeof Text> = {
  title: 'Library/Atoms/Text',
  component: Text,
  parameters: {
    layout: 'centered',
  },
  decorators: [
    Story => (
      <Theme>
        <Story />
      </Theme>
    ),
  ],
  argTypes: {
    children: {
      control: 'text',
    },
    size: {
      control: {
        type: 'object',
      },
    },
    weight: {
      control: {
        type: 'object',
      },
    },
    className: {
      control: 'text',
    },
  },
  args: {
    children: 'Sample Text',
    size: '3',
    weight: 'regular',
    as: 'span',
  },
};

export default meta;

type Story = StoryObj<TextProps>;

export const BreakPointVariantText: Story = {
  args: {
    size: { initial: '1', xs: '2', sm: '3', md: '4', lg: '4', xl: '6' },
    weight: { initial: 'regular', sm: 'semibold', md: 'bold' },
    children: 'Custom Text with Breakpoint Variant Styles',
    className: 'custom-class',
  },
};
