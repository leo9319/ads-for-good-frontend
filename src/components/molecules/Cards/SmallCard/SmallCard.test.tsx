import React from 'react';
import { render } from '@testing-library/react';
import Text from '@radix-styles/atoms/Text';
import SmallCard from './SmallCard';

describe('Small Card', () => {
  it('Should Primary mode with size 1 render successfully', () => {
    const { baseElement } = render(
      <SmallCard size="1" mode="primary">
        <Text>
          {
            'World Vision Canada is a Christian relief, development, and advocacy organization working to create lasting change in the lives of children and families. (To accommodate up to five lines of text).'
          }
        </Text>
      </SmallCard>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('Should inverse mode with Size 2 render successfully', () => {
    const { baseElement } = render(
      <SmallCard size="2" mode="inverse">
        <Text>
          {
            'World Vision Canada is a Christian relief, development, and advocacy organization working to create lasting change in the lives of children and families. (To accommodate up to five lines of text).'
          }
        </Text>
      </SmallCard>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
