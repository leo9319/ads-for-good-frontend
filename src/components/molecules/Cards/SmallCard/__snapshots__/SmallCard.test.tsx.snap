// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Card Should Primary mode with size 1 render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-reset rt-BaseCard rt-Card rt-r-size-1 rt-variant-surface container size-1 primary"
    >
      <span
        class="rt-Text mocked-class-name typography-font-family-default"
        id=""
      >
        World Vision Canada is a Christian relief, development, and advocacy organization working to create lasting change in the lives of children and families. (To accommodate up to five lines of text).
      </span>
    </div>
  </div>
</body>
`;

exports[`Data Card Should inverse mode with Size 2 render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-reset rt-BaseCard rt-Card rt-r-size-1 rt-variant-surface container size-2 inverse"
    >
      <span
        class="rt-Text mocked-class-name typography-font-family-default"
        id=""
      >
        World Vision Canada is a Christian relief, development, and advocacy organization working to create lasting change in the lives of children and families. (To accommodate up to five lines of text).
      </span>
    </div>
  </div>
</body>
`;

exports[`Small Card Should Primary mode with size 1 render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-reset rt-BaseCard rt-Card rt-r-size-1 rt-variant-surface container size-1 primary"
    >
      <span
        class="rt-Text mocked-class-name typography-font-family-default"
        id=""
      >
        World Vision Canada is a Christian relief, development, and advocacy organization working to create lasting change in the lives of children and families. (To accommodate up to five lines of text).
      </span>
    </div>
  </div>
</body>
`;

exports[`Small Card Should inverse mode with Size 2 render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-reset rt-BaseCard rt-Card rt-r-size-1 rt-variant-surface container size-2 inverse"
    >
      <span
        class="rt-Text mocked-class-name typography-font-family-default"
        id=""
      >
        World Vision Canada is a Christian relief, development, and advocacy organization working to create lasting change in the lives of children and families. (To accommodate up to five lines of text).
      </span>
    </div>
  </div>
</body>
`;
